cmd/collector/collector.go

```diff
@@ -138,20 +138,10 @@ func run() error {
 	messageReceived := make(chan *entities.Message)
 	go func() {
 		go cp.Start()
-		msgCount := 0
-		// This can be simplified if collector uses a message channel.
-		// Are there any use cases for having list of messages in collector process.
-		for {
-			ipfixMessages := cp.GetMessages()
-			currentLen := len(ipfixMessages)
-			if currentLen > msgCount {
-				for i := msgCount; i < currentLen; i++ {
-					klog.Info("Processing IPFIX message")
-					messageReceived <- ipfixMessages[i]
-					msgCount++
-				}
-			}
-			time.Sleep(5 * time.Millisecond)
+		msgChan := cp.GetMsgChan()
+		for message := range msgChan {
+			klog.Info("Processing IPFIX message")
+			messageReceived <- message
```

Comment:
Just curious, why not using the channel from cp.GetMsgChan() directly inside signalHandler instead of piping one channel to another channel?

For example, can we extend signalHandler (or rename it to eventHandler) as

func eventHandler(stopCh chan struct{}, cp *collectingProcess) {
	cp.Start()
	msgChan := cp.GetMsgChan()
	signalCh := make(chan os.Signal, 1)
	signal.Notify(signalCh, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)

        for {
                select {
                case msg := <-msgChan:
                        printIPFIXMessage(msg)
                case <-signalCh:
                        close(stopCh)
                        return
                }
        }
}

Then there is no need for the go func on lines 139-146?
Thanks.

Reply:
Hi Shihao,
Yes, I noticed that too. We do not need an extra message channel. Thought of taking care of this in follow-up patch as we want to move on with v0.3.0 release to have this available for changing Antrea e2e tests as part of Antrea 0.11.0 release.

However, I think it is best to have a separate signal handler that sends out signal on stopCh. This way we can have multiple independent go routines that start their functionalities (using `Run` as in convention of Antrea or other well known go applications) and they can be stopped with the stopCh. Extending this collector application will be easier.

Reply:
Yes, a dedicated signal handler sounds good too. Then we can have the original go func (i.e. lines 139-146) calling printIPFIXMessage on the messages from the message channel directly. Is that right? Thanks.

Reply:
Yes that's correct. It taken in stopCh as argument.

Reply:
Btw, please feel free to raise a PR against release-0.3.0 branch. We can add this to patch release.